Frameworks/Foundation/NSOperation.mm

```diff
@@ -344,14 +347,14 @@ - (NSArray*)dependencies {
  @Status Interoperable
 */
 - (void)dealloc {
-    { // _dependenciesLock scope
-        std::lock_guard<std::recursive_mutex> lock(_dependenciesLock);
-        for (NSOperation* op in (NSSet*)_dependencies) {
-            [op removeObserver:self forKeyPath:@"isFinished" context:(void*)NSOperationContext];
-        }
+    for (NSOperation* op in (NSSet*)_dependencies) {
+        [op removeObserver:self forKeyPath:@"isFinished" context:(void*)NSOperationContext];
+    }
```

Comment:
this ends up being a no-op if _dependencies is nil, so it's redundant to check for it. additionally, I would prefer to keep the check strongly tied to the self-observation, as it's the only thing that's actually causing problems